src/sql/workbench/api/browser/mainThreadDataProtocol.ts

```diff
@@ -525,7 +525,7 @@ export class MainThreadDataProtocol extends Disposable implements MainThreadData
 			generateScript(tableInfo: azdata.designers.TableInfo): Thenable<string> {
 				return self._proxy.$generateScriptForTableDesigner(handle, tableInfo);
 			},
-			generatePreviewReport(tableInfo: azdata.designers.TableInfo): Thenable<string> {
+			generatePreviewReport(tableInfo: azdata.designers.TableInfo): Thenable<string | { report: string, format: string }> {
```

Comment:
And then I'd also suggest making the format a const enum value instead - much easier to work with and clearly defines what the expected values are.